src/App.tsx

```diff
@@ -21,75 +22,109 @@ enum SortType {
   LENGTH,
 }
 
-type ReorderOptions = {
-  sortType: SortType,
+type State = {
   isReversed: boolean,
+  sortType: SortType,
 };
 
-// Use this function in the render method to prepare goods
-export function getReorderedGoods(
-  goods: string[],
-  { sortType, isReversed }: ReorderOptions,
-) {
-  // To avoid the original array mutation
-  const visibleGoods = [...goods];
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  sortByAlphabet = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
 
-  return visibleGoods;
-}
+  sortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+  reverse = () => {
+    this.setState(state => (
+      { isReversed: !state.isReversed }
+    ));
+  };
+
+  reset = () => {
+    this.setState({ sortType: SortType.NONE, isReversed: false });
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+    const visibleGoods = [...goodsFromServer];
+
+    switch (sortType) {
+      case SortType.ALPHABET:
+        visibleGoods.sort((g1, g2) => (
+          g1.localeCompare(g2)
+        ));
+        break;
+
+      case SortType.LENGTH:
+        visibleGoods.sort((g1, g2) => (
+          g1.length - g2.length
+        ));
+
+        break;
+
+      default:
+        break;
+    }
+
+    if (this.state.isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            onClick={this.sortByAlphabet}
+            type="button"
+            className={cn('button', 'is-info',
+              { 'is-light': sortType !== SortType.ALPHABET })}
+          >
```

Comment:
no need in comma here
```suggestion
            className={cn('button is-info',
              { 'is-light': sortType !== SortType.ALPHABET })}
          >
```